### DISCUSSION TITLE
Ferramentas/Frameworks para desenvolvimento rápido de MVCs e POCs

### DISCUSSION THREAD

**Discussion by brunoventura** [frontendbr/forum#637] [Perguntas e Respostas]
Bom dia galera,

Atualmente estou me envolvendo em vários projetos onde preciso desenvolver POCs e MVPs com um certo nível de velocidade para validar ideias e iterar o mais rápido possível, e estou atrás de ferramentas que me auxiliem com isso.

Costumo já usar react e angular4 em projetos mais duradouros, porém para esse caso gostaria de algo mais quick win. Dei uma revisitada no [Meteor](https://www.meteor.com/), porém o conheci e testei fazem alguns anos, porém não sei como está o envelhecimento dele, estou considerando usá-lo.

Algumas premissas:
1. The more JS, the better
2. Preferencialmente integrável com TS ou flow (gosto pessoal)
3. Que tenha blocos prontos para utilizar (ex.: login, user management)
4. Não necessita ser extremamente confiável nem extensível, o objetivo não é fazer um poc-2-prd

Alguns questionamentos:
1. Alguém anda usando meteor? Quer compartilhar experiências, prós e contras?
2. Alguém já integrou meteor com TS ou Flow?
3. Alguém conhece alguma alternativa tão boa quanto ou melhor?
4. Alguém conhece formas de ter uma produtividade tão boa quanto ou melhor com React ou Angular4 (ou qualquer outra ferramenta)

**Comment by caioincau** [comment]
3. Cara, eu iria de Vue.js, ele tem uma curva de aprendizado minima, tem cli para fazer o scaffold do projeto, inclusive ele nasceu dentro da google, como uma ferramenta para prototipação.

**Comment by brunoventura** [comment]
@icaioincau já usei o vue quando tava em beta ainda, curti, sei que não é o foco do grupo, mas o que você usaria pra integrar com backend, questão de user management, login e afins? Algo _bootstrapzável_.

Acredita que vue+algo seria melhor que meteor pra um desenvolvimento frontend+backend?

**Comment by guilhermewaess** [comment]
Eai @brunoventura, eu gosto bastante do Vue! 
Mas lendo a premissa ali de ser "integrável com TS", imho, me parece ainda vai levar um tempo pro Vue ficar bom com TS. 
Tem algumas alternativas de autenticação pra ele, é só escolher o que se encaixa melhor.

**Comment by weltonsanches** [comment]
@brunoventura  Não manjo de vue. Mas é o próximo framework que vou estudar. Uso Laravel em PHP no back-end e existe uma infinidade de tutorial e projetos desenvolvidos integrando os 2.

**Comment by ghost** [comment]
O [angular@cli](https://cli.angular.io/) também é ótimo para criar um flow que acelera o desenvolvimento de protótipos.

```shellscript
ng new <project-name>
ng generate component <name-component>
ng serve
```

***

O [Vue cli](https://github.com/vuejs/vue-cli) muda um pouco o conceito, tem apenas scaffold do inicio de um projeto.
```shellscript
vue init <template-name> <project-name>
```

Porém, a forma dos componentes .vue é bem estruturada para você escrever protótipos e componentes de uma forma centralizada.

```html
<template lang="pug">
</template>
<script></script>
<style lang="stylus">
</style>
```
Sem falar dos vue-loader, onde possibilita você integrar fácilmente templates engines, pre-processadores e transpilers.

***

## Opnião
Acho que deve experimentar o máximo de frameworks para ver qual te dará mais perfomance.
Existe diversas formas de chegar ao mesmo resultado.

**Comment by VitorLuizC** [comment]
Firebase é incrível pra fazer os serviços de Login, gerenciamento de banco de dados, storage e um monte de outras coisas. Hoje meus projetos pessoais tem todo o back-end no Firebase e o front-end em Vue/Vuex e algumas paradinhas mais...

**Comment by onlurking** [comment]
Pra prototipar um frontend, o [json-server](https://github.com/typicode/json-server) como API/backend é uma mão gigantesca na roda.

**Comment by cezarpretto** [comment]
Eu iria de Vue com toda a certeza. Já trabalhei com ng1, ng2 e Vue. Sem dúvidas, hoje considero Vue a melhor alternativa. Não vejo necessidade de TS (JS puro é lindo), mas também é um gosto pessoal.

**Comment by victorkurauchi** [comment]
@brunoventura , existe um projeto muito interessante que agiliza muito (backend e frontend) quando você está iniciando o projeto. Acredito que atende todas suas premissas, inclusive a de login e user management.

http://keystonejs.com/

O projeto é um CMS e te disponibiliza o frontend em React. Não sei se isso é um bloqueador pra ti, mas ele trabalha com o mongodb out ot the box, não tentei trabalhar com ele com algum outro db. Espero que ajude.
